engine/swarm/store_sensitive_strings.md

```diff
@@ -0,0 +1,234 @@
+                            -e MYSQL_ROOT_PASSWORD=/run/secrets/mysql_password \
+                            mysql/mysql-server
+    ```
+
+4.  Verify that the `mysql` service is running using `docker service ls`.
+
+    ```bash
+    $ docker service ls
+
+    ID            NAME   MODE        REPLICAS  IMAGE
+    6xw6rke8k1ul  mysql  replicated  1/1       mysql/mysql-server@sha256:406a4256f5092f88e0c424a0749d1258eb25f5db14eeed4d46532fbda3c63266
+    ```
+
+5.  Now that MySQL is set up, you can create a WordPress container that connects
+    to the MySQL server. The WordPress service has the following characteristics:
+
+    - Joins the service to the `wp_network` overlay network, and resolves to the
+      DNS name `wordpress`.
+    - Because the scale is set to 1, only a single WordPress task runs.
+      Load-balancing WordPress is left as an exercise to the reader, because of
+      limitations with storing WordPress session data on the container filesystem.
+    - Exposes WordPress on port 30000 of the host machine, so that you can access
+      it from external hosts. You could expose port 80 if you do not have a web
+      server running on port 80 already.
+    - Grants the service access to the `mysql_password` secret, which will be
+      available as plain-text within the container at `/run/secrets/mysql_password`.
+    - Sets the environment variable `WORDPRESS_DB_PASSWORD` to this file path. The
+      `mistysj/wordpress` image is a fork of the official WordPress image which
+      was created for this example and checks to see if the password is a path
+      to a file, and if so, reads the password string from that file.
+
+    ```bash
+    $ docker service create --name="wordpress" \
+                            --network="wp_network" \
+                            --replicas="1" \
+                            --publish 30000:80 \
+                            --secret=mysql_password \
+                            -e WORDPRESS_DB_PASSWORD="/run/secrets/mysql_password" \
+                            -e WORDPRESS_DB_HOST="mysql:3306" \
+                            mistysj/wordpress
+    ```
+
+6.  Access `http://localhost:30000/` from any Swarm node and begin using
+    WordPress.
+
+
+## Build support for Docker Secrets into your images
+
+If you develop a container that can be deployed as a service and accepts a
+sensitive string, such as a credential, as an environment variable, you can make
+use of Docker secrets. The following examples show variations on one approach.
+
+**Note**: These examples use syntax specific to shells like Bash. If you use a
+different shell, verify that shell's syntax for checking for the existence of
+a file.
+
+When evaluating the value of the environment variable,
+first check whether that value corresponds to a file path available within the
+container read the credentail from the file if so. The `mistysj/wordpress` image
+adds the following check to see if the value of `WORDPRESS_DB_PASSWORD` is a
```

Comment:
We should probably document why we don't support environment variables in the first place. They are less secure than files.

See: https://github.com/docker/docker/pull/9176#issuecomment-99542089